install-poetry.py

```diff
@@ -755,7 +755,7 @@ def _compare_versions(x, y):
                 colorize("error", "Version {} does not exist.".format(self._version))
             )
 
-            return None, None
+            sys.exit(1)
```

Comment:
Thanks for the contribution. Not an official contributor, but my 2 cents:

It's generally frowned upon to use `sys.exit` in the middle of the code. This function is `get_version`, it should return the version, not exit if the version doesn't exist. An easy example of when this would be bad is in case you'd want to create unit tests for this method. You want to be able to check what happens when the version does not exist, and if you're exiting the run in the middle of the test it wouldn't work.
If you look closer, you can see in the calling code (`run`) there is already a place dealing with the case where `get_version` returns `None`. However, there it returns 0 from `run` instead of 1. The return value of `run` is the return value of `main`, which then `sys.exit`s, at the only correct spot in the script.
So the better fix here would be changing the return value of `run` to 1 instead of 0 when the version is None.

Reply:
Thanks for your feedback!

I agree that using `sys.exit` is generally not a great look. However, I also don't really like returning `None, None` there; it doesn't signal nicely that something went wrong. Raising an exception makes more sense to me. But then the code calling `get_version` needs to catch this, and subsequently handle it. Since `sys.exit` basically just throws an exception and propagates all the way up, I thought it was actually quite an elegant way of handling it. It's a bit of a nightmare to write tests for, but since there's no tests... :)

That being said, I see your point, and I see two viable alternatives:

1. Raise a `ValueError` instead of returning `None, None`, and then handle this with a `try/except` around the `get_version` call, returning `1` in case of an exception.
2. Do not change the `get_version` method, instead follow the `get_version` call with a check whether the returned values are `None, None`, and return `1`.

So I think you are proposing version 2?

Reply:
I slept on it, and I agree that your proposal is more in line with the spirit of the existing code.

The `run` function will now return `1` if both `version` and `current_version` are `None`; which is the the case only when the specified version does not exist.

Reply:
Actually, after looking at it some more, I went with my option 1. You can imagine some edge cases where `None, None` is returned even if there was no invalid version error. Now by raising an exception, the code is being very It is the most explicit.

Reply:
I agree. Looking at it again, the return values of `get_version` are a bit wonky, sometimes returning `None` when there isn't an error. Raising an exception seems like a good idea, even if it's a bit different from the current structure of the code.